dev/ci/docker/bionic_coq/Dockerfile

```diff
@@ -7,13 +7,19 @@ LABEL maintainer="e@x80.org"
 ENV DEBIAN_FRONTEND="noninteractive"
 
 RUN apt-get update -qq && apt-get install --no-install-recommends -y -qq \
+        # Dependencies of the test-suite and of external projects
         m4 automake autoconf time wget rsync git gcc-multilib opam \
+        # Dependencies of CoqIDE
         libgtk2.0-dev libgtksourceview2.0-dev \
-        texlive-latex-extra texlive-fonts-recommended texlive-science tipa \
-        python3-sphinx python3-pexpect python3-sphinx-rtd-theme python3-bs4 python3-sphinxcontrib.bibtex \
-        python3-setuptools python3-wheel python3-pip
-
-RUN pip3 install antlr4-python3-runtime
+        # Dependencies of stdlib and sphinx doc
+        texlive-latex-extra texlive-fonts-recommended texlive-xetex \
+        latexmk xindy python3-pip python3-setuptools \
+        # Dependencies of source-doc and coq-makefile
+        texlive-science tipa
+
+# More dependencies of the sphinx doc
+RUN pip3 install sphinx==1.7.0 sphinx_rtd_theme beautifulsoup4 \
+        antlr4-python3-runtime pexpect sphinxcontrib-bibtex
```

Comment:
For example I don't see a reason you install `pexpect` using pip. Similarly for beautifuldsoup4.